.eslintrc.js

```diff
@@ -11,36 +11,42 @@ module.exports = {
     browser: true,
     node: true,
   },
-  extends: ['plugin:import/recommended', 'airbnb', 'prettier', 'prettier/react'],
-  parser: 'babel-eslint',
+  extends: [
+    'plugin:import/recommended',
+    'plugin:import/typescript',
+    'airbnb-typescript',
+    'prettier',
+    'prettier/react',
+    'prettier/@typescript-eslint',
+  ],
+  parser: '@typescript-eslint/parser',
   parserOptions: {
-    ecmaVersion: 7,
-    sourceType: 'module',
+    // project: './tsconfig.json',
   },
-  plugins: ['babel', 'material-ui', 'react-hooks'],
+  plugins: ['material-ui', 'react-hooks', '@typescript-eslint'],
   settings: {
     'import/resolver': {
       webpack: {
         config: path.join(__dirname, './docs/webpackBaseConfig.js'),
       },
     },
   },
-  /**
-   * Sorted alphanumerically within each group. built-in and each plugin form
-   * their own groups.
-   */
+  // Sorted alphanumerically
   rules: {
+    '@typescript-eslint/dot-notation': 'off', // Too slow
+    '@typescript-eslint/no-implied-eval': 'off', // Too slow
+    '@typescript-eslint/no-throw-literal': 'off', // Too slow
     'consistent-this': ['error', 'self'],
-    'linebreak-style': 'off', // Doesn't play nicely with Windows
-    // just as bad as "max components per file"
-    'max-classes-per-file': 'off',
-    'no-alert': 'error',
-    // Strict, airbnb is using warn; allow warn and error for dev environments
-    'no-console': ['error', { allow: ['warn', 'error'] }],
-    'no-constant-condition': 'error',
-    // Airbnb use error
-    'no-param-reassign': 'off',
-    'no-prototype-builtins': 'off',
+    'import/export': 'off', // Not sure why it doesn't work
```

Comment:
We could simply enable all `import/*` rules only for `.js` foles
```js
 "overrides": [
        {
            "files": ["*.js"],
            "processor": { 
                 "import/export": "error"
            }
       }
    ]
```

Reply:
If we need to disable rules then I'd put this effort off until the issues are resolved. If linting is not important then we can have a discussion about that. But disabling rules just because we can't get them to work defeats the purpose of these rules.

Reply:
> But disabling rules just because we can't get them to work defeats the purpose of these rules.

It's a tradeoff, one step at the time. @dmtrKovalenko great idea

Reply:
> It's a tradeoff, one step at the time.

It's not worth it to merge such a big PR if we have to revert it later.

Reply:
> It's not worth it to merge such a big PR if we have to revert it later.

I have scoped `'import/export': 'off'` to the `.d.ts` files only.